src/Avalonia.Controls/Control.cs

```diff
@@ -53,24 +55,48 @@ public class Control : InputElement, IControl, INamed, IVisualBrushInitialize, I
         /// Event raised when an element wishes to be scrolled into view.
         /// </summary>
         public static readonly RoutedEvent<RequestBringIntoViewEventArgs> RequestBringIntoViewEvent =
-            RoutedEvent.Register<Control, RequestBringIntoViewEventArgs>("RequestBringIntoView", RoutingStrategies.Bubble);
+            RoutedEvent.Register<Control, RequestBringIntoViewEventArgs>(
+                "RequestBringIntoView",
+                RoutingStrategies.Bubble);
 
         /// <summary>
         /// Provides event data for the <see cref="ContextRequested"/> event.
         /// </summary>
         public static readonly RoutedEvent<ContextRequestedEventArgs> ContextRequestedEvent =
-            RoutedEvent.Register<Control, ContextRequestedEventArgs>(nameof(ContextRequested),
+            RoutedEvent.Register<Control, ContextRequestedEventArgs>(
+                nameof(ContextRequested),
                 RoutingStrategies.Tunnel | RoutingStrategies.Bubble);
         
+        /// <summary>
+        /// Defines the <see cref="Loaded"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> LoadedEvent =
+            RoutedEvent.Register<Control, RoutedEventArgs>(
+                nameof(Loaded),
+                RoutingStrategies.Direct);
+
+        /// <summary>
+        /// Defines the <see cref="Unloaded"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> UnloadedEvent =
+            RoutedEvent.Register<Control, RoutedEventArgs>(
+                nameof(Unloaded),
+                RoutingStrategies.Direct);
+
         /// <summary>
         /// Defines the <see cref="FlowDirection"/> property.
         /// </summary>
         public static readonly AttachedProperty<FlowDirection> FlowDirectionProperty =
-            AvaloniaProperty.RegisterAttached<Control, Control, FlowDirection>(nameof(FlowDirection), inherits: true);
+            AvaloniaProperty.RegisterAttached<Control, Control, FlowDirection>(
+                nameof(FlowDirection),
+                inherits: true);
 
+        private bool _isLoaded = false;
+        private bool _isLoadedProcessingRequested = false;
         private DataTemplates? _dataTemplates;
         private IControl? _focusAdorner;
         private AutomationPeer? _automationPeer;
+        private HashSet<Control> _loadedQueue = new HashSet<Control>();
```

Comment:
This isn't supposed to be instance field, isn't it?
If 10 controls are attached to the visual tree, then they should be added to the shared queue (static?) and Loaded in single dispatcher job.

Reply:
No, it definitely is not...

Reply:
Fixed, thanks